src/qibosoq/programs/base.py

```diff
@@ -272,11 +272,28 @@ def collect_shots(self) -> Tuple[list, list]:
 
         unique_adcs, adc_count = np.unique(adcs, return_counts=True)
 
+        number_of_adcs = len(unique_adcs)
+        len_buf = len(self.di_buf[0])
+
         for idx, adc_ch in enumerate(unique_adcs):
-            count = adc_count[idx]
-            shape = (count, self.expts, self.reps) if self.expts else (count, self.reps)
-            i_val = self.di_buf[idx].reshape(shape) / lengths[idx]
-            q_val = self.dq_buf[idx].reshape(shape) / lengths[idx]
+            count = int(adc_count[idx])
+            try:
+                shape = (
+                    count,
+                    int(np.prod(self.sweep_axes)),
+                    self.reps,
+                )  # type: Union[Tuple[int, int], Tuple[int, int, int]]
+            except AttributeError:
+                shape = (count, self.reps)
```

Comment:
Could you add a comment here to explain what is happening (and why you might not have `self.sweep_axes`).

I believe you're inheriting this nice inhomogeneity from `QickProgram` (I didn't check myself). But it would have been nicer if `self.sweep_axes`, for no sweepers, would have just been an empty sequence.

Reply:
I know that this is a bit weird, I could check if self is NDAveragerProgram, maybe.
Anyway, I added some more specific comment. Something that could be done in the future is unify the shapes for all the different types of acquisition. However this will require some more changes and I would do that in a new release.